File: JavaScript projects/JavaScript Basics/Conditional Statements Advanced/Exercises/06_operations_between_numbers.js
function operations(numberOne, numberTwo, operator){
    let result;
    let parity;
    let division_is_valid = true;
    switch (operator){        
        case "+":
            result = numberOne + numberTwo;
            break;
        case "-":
            result = numberOne - numberTwo;
            break;
        case "*":
            result = numberOne * numberTwo;
            break;
        case "/":
            if (numberTwo != 0){
            result = numberOne * numberTwo;
            }else{
                result = 0;
                division_is_valid = false;
            }
            break;
        case "%":
            result = numberOne % numberTwo;
            break;
    }
    if (operator === "+" || operator === "-" || operator === "*"){
        if (result % 2 === 0){
            parity = "even";
        }else{
            parity = "odd";
        }
        console.log(`${numberOne} ${operator} ${numberTwo} = ${result} - ${parity}`);
    }else if(operator === "%"){
        console.log(`${numberOne} ${operator} ${numberTwo} = ${result}`);
    }else if(division_is_valid){
        console.log(`${numberOne} ${operator} ${numberTwo} = ${(result.toFixed(2))}`);
    }else{
        console.log(`Cannot divide ${numberOne} by zero`);
    }
}
operations(10,12,"+");

File: Python projects/Python Basics/03_conditional_statements_advanced/exercises/06_operations_between_numbers.py
number1 = int(input())
number2 = int(input())
operator = str(input())

result = 0

if operator == '+' or operator == '-' or operator == '*':
    if operator == '+':
        result = number1 + number2
    elif operator == '-':
        result = number1 - number2
    else:
        result = number1 * number2
    if result % 2 == 0:
        print(f"{number1} {operator} {number2} = {result} - even")
    else:
        print(f"{number1} {operator} {number2} = {result} - odd")
elif operator == '/' or '%':
    if number2 == 0:
        print(f"Cannot divide {number1} by zero")
    elif operator == '/':
        result = number1 / number2
        print(f"{number1} {operator} {number2} = {result:.2f}")
    else:
        result = number1 % number2
        print(f"{number1} {operator} {number2} = {result}")
